keras/backend/cntk_backend.py

```diff
@@ -1434,7 +1475,12 @@ def _recurrence(x, states, m):
                 uses_learning_phase = True
 
             if m is not None:
-                new_states = [C.element_select(m, n, s) for n, s in zip(new_states, past_values)]
+                new_states_temp = []
+                for n, s in zip(new_states, past_values):
+                    new_states_temp.append(C.element_select(m, n, s))
+
+                new_states = new_states_temp
```

Comment:
There is no need for a temporary variable. You can write `new_states = []` directly.

Reply:
This suggestion has been incorporated.

Reply:
The `zip` command is using the variable `new_states` in the for loop. So I think I need to incorporate a temporary variable again to make it correct.

Reply:
Oups, sorry, my bad. You are right! Please proceed with the temporary variable.

Reply:
Thanks for the clarifications.